src/UglyToad.PdfPig/Content/BasePageFactory.cs

```diff
@@ -136,15 +136,15 @@ public abstract class BasePageFactory<TPage> : IPageFactory<TPage>
                         throw new InvalidOperationException($"Could not find the contents for object {obj}.");
                     }
 
-                    bytes.AddRange(contentStream.Decode(FilterProvider, PdfScanner));
+                    bytes.Write(contentStream.Decode(FilterProvider, PdfScanner).Span);
 
                     if (i < array.Data.Count - 1)
                     {
-                        bytes.Add((byte)'\n');
+                        bytes.Write((byte)'\n');
                     }
                 }
 
-                page = ProcessPageInternal(number, dictionary, namedDestinations, mediaBox, cropBox, userSpaceUnit, rotation, initialMatrix, bytes);
+                page = ProcessPageInternal(number, dictionary, namedDestinations, mediaBox, cropBox, userSpaceUnit, rotation, initialMatrix, bytes.WrittenSpan.ToArray());
```

Comment:
I assume you will remove the `ToArray()` in later commit

Reply:
Eliminated! This was potentially a LOH allocation, so great to kill.